README.md

```diff
@@ -1,11 +1,57 @@
+  * ```VK_COLOR_SPACE_SRGB_NONLINEAR_KHR``` is now ```vk::ColorSpaceKHR::eSrgbNonlinear```.
+  *  ```VK_STRUCTURE_TYPE_PRESENT_INFO_KHR``` is now ```vk::StructureType::ePresentInfoKHR```.
+*  Flag bits are handled like scoped enums with the addition that the _BIT suffix has also been removed.
+
+# Handles
+Vulkan-Hpp declares a class for all handles to ensure full type safety and to add support for member functions on handles. A member function has been added to a handle class for each
+ function which accepts the corresponding handle as first parameter. Instead of ```vkBindBufferMemory(device, ...)``` one can write ```device.bindBufferMemory(...)``` or ```vk::bindBufferMemory(device, ...)```.
+
+# C/C++ Interop for Handles
+On 64-bit platforms Vulkan-Hpp supports implicit conversions between C++ Vulkan handles and C Vulkan handles. On 32-bit platforms all non-dispatchable handles are defined as ```uint64_t``` and thus not typesafe. 
+Due to that Vulkan-Hpp does not enable implicit conversion for 32-bit platforms by default and it is recommended to use a static_cast for the conversion like this: ```VkDevice = static_cast<VkDevice>(cppDevice)```
+to prevent converting some arbitrary int to a handle or vice versa by accident. If you're developing your code on a 64-bit platform, but want compile your code for a 32-bit platform without adding the explicit casts
+you can define ```VULKAN_HPP_TYPESAFE_CONVERSION``` to 1 in your build system or before including ```vulkan.hpp```.
+
+# Flags
+The scoped enum feature adds type safety to the flags, but also prevents using the flag bits as input for bitwise operations like & and |. 
+As solution Vulkan-Hpp provides a template class ```vk::Flags``` which brings the standard operations like &=, |=, & and | to our scoped enums. Except for the initialization with 0 this class behaves exactly like a normal bitmask
+ with the improvement that it is impossible to set bits not specified by the corresponding enum by accident. Here are a few examples for the bitmask handling:
 
-Vulkan is a C API and as such inherits all common pitfalls of using a general C programming library. The motivation of a low-level Vulkan C++ API is to avoid these common pitfalls by applying commonly known C++ features while keeping the overall structure of a Vulkan program and preserving the full freedom it provides as low-level graphics API. An additional guideline we followed was not to introduce additional runtime overhead by providing a header-only library with inline functions.
+```c++
+vk::ImageUsage iu1; // initialize a bitmask with no bit set
+vk::ImageUsage iu2 = {}; // initialize a bitmask with no bit set
+vk::ImageUsage iu3 = vk::ImageUsage::eColorAttachment; // initialize with a single value
+vk::ImageUsage iu4 = vk::ImageUsage::eColorAttachment | vk::ImageUsage::eStorage; // or two bits to get a bitmask
+PipelineShaderStageCreateInfo ci( {} /* pass a flag without any bits set, ...);
```

Comment:
Up to a second ago I didn't have a solution for this problem. 

Now I'm wondering if it'd be acceptable to provide a constructor which accepts a nullptr_t. This way one can pass a constant 0 to flags. The downside is one can also pass a nullptr to create a flags object with no bits set. Is this too confusing?